flake.nix

```diff
@@ -1,58 +1,118 @@
 {
   inputs = {
-    naersk.url = "github:nix-community/naersk/master";
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
     utils.url = "github:numtide/flake-utils";
     nixtract.url = "github:tweag/nixtract";
   };
 
   outputs =
     { self
+    , crane
     , nixpkgs
     , utils
-    , naersk
     , nixtract
     }:
     utils.lib.eachDefaultSystem
       (system:
       let
         pkgs = import nixpkgs { inherit system; };
-        naersk-lib = pkgs.callPackage naersk { };
+        crane-lib = crane.lib.${system};
         cyclonedx = pkgs.callPackage ./nix/cyclonedx.nix { };
         nixtract-cli = nixtract.defaultPackage.${system};
+
+        # Here we start the crane stuff
+        common-crane-args = {
+          pname = "genealogos";
+          src = crane-lib.cleanCargoSource (crane-lib.path ./.);
+          strictDeps = true;
+
+          cargoArtifacts = cargo-artifacts;
```

Comment:
Although it seems to work in practic , the cyclic dependency between `common-crane-args` and `cargo-artifacts` looks a bit surprising at first.